src/mscorlib/src/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -334,10 +325,6 @@ public override long Position
                     throw new NotSupportedException(SR.NotSupported_UmsSafeBuffer);
                 if (!_isOpen) __Error.StreamIsClosed();
 
-                // Note: subtracting pointers returns an Int64.  Working around
```

Comment:
Another example:

```
static void Main() {
     var ums = new UnmanagedMemoryStream((byte*)0x40000000, 1);
     ums.PositionPointer = (byte*)0xF0000000;
     Console.WriteLine(ums.Position.ToString());
}
```

If you run it on 32-bit machine, the current implementation will produce negative position. Instead, it should produce Position == 0xB0000000. We should add this one as regression test case to corefx.